static/app/views/dashboards/detail.tsx

```diff
@@ -594,6 +614,24 @@ class DashboardDetail extends Component<Props, State> {
     );
   };
 
+  /* Handles POST request for Edit Access Selector Changes */
+  onChangeEditAccess = (newDashboardPermissions?: DashboardPermissions) => {
+    if (!defined(newDashboardPermissions)) {
+      return;
+    }
+    const {dashboard, api, organization} = this.props;
+
+    const dashboardCopy = cloneDashboard(dashboard);
+    dashboardCopy.permissions = newDashboardPermissions;
+
+    updateDashboard(api, organization.slug, dashboardCopy).then(
+      (newDashboard: DashboardDetails) => {
+        addSuccessMessage(t('Dashboard Edit Access updated.'));
+        return newDashboard;
```

Comment:
In other places of this file, e.g. `onCommit` we use `onDashboardUpdate` to set the dashboard in the parent, does this solve the issue of passing the state down properly? I don't think you'll need to update the state because there shouldn't be a "modified dashboard" when this accessible